common/download.go

```diff
@@ -195,9 +197,9 @@ func (*HTTPDownloader) Cancel() {
 	// TODO(mitchellh): Implement
 }
 
-func (d *HTTPDownloader) Download(dst io.Writer, src *url.URL) error {
+func (d *HTTPDownloader) Download(dst *os.File, src *url.URL) error {
 	log.Printf("Starting download: %s", src.String())
-	req, err := http.NewRequest("GET", src.String(), nil)
+	req, err := http.NewRequest("HEAD", src.String(), nil)
```

Comment:
Not all servers support HEAD. I think if this is the case, we should just fall back to using `GET` with no range.